src/lesson3/homework03/IDequeImpl.java

```diff
@@ -0,0 +1,62 @@
+package lesson3.homework03;
+
+import lesson3.queue.QueueImpl;
+
+public class IDequeImpl<E> extends QueueImpl<E> implements IDeque<E> {
+    public IDequeImpl(int maxSize) {
+        super(maxSize);
+    }
+
+    @Override
+    public boolean addFirst(E value) {
+        if (isFull()) {
+            return false;
+        }
+
+        if (head == 0) {
+            head = data.length - 1;
+        }
+
+        data[head] = value;
```

Comment:
```suggestion
        data[head--] = value;
```
не хватает кода по модификации состояния метки